setup.py

```diff
@@ -0,0 +1,119 @@
+        except OSError:
+            raise RuntimeError("CMake must be installed to build the extensions.")
+
+        # build extensions
+        for ext in self.extensions:
+            self.build_extension(ext)
+
+
+    def build_extension(self, ext: CMakeExtension) -> None:
+        """Execute CMake and build extension."""
+        # get full path of the extension
+        ext_fullpath = Path.cwd() / self.get_ext_fullpath(ext.name)
+        extdir = ext_fullpath.parent.resolve()
+
+        # get debug/release configuration
+        debug = int(os.environ.get("DEBUG", 0)) if self.debug is None else self.debug
+        cfg = "Debug" if debug else "Release"
+        cmake_args = [
+            f"-DCMAKE_LIBRARY_OUTPUT_DIRECTORY={extdir}{os.sep}",
+            f"-DCMAKE_BUILD_TYPE={cfg}",
+            f"-DOPENSN_WITH_LUA=OFF",
+            f"-DOPENSN_WITH_PYTHON=ON"
+        ]
+        if sys.platform.startswith("win"):
+            cmake_args += [f"-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{cfg.upper()}={extdir}"]
+
+        # get generator from environment if precised
+        cmake_generator = os.environ.get("CMAKE_GENERATOR", "")
+        if cmake_generator:
+            cmake_args += ["-G {cmake_generator}"]
+        # get extra CMake arguments
+        if "CMAKE_ARGS" in os.environ:
+            cmake_args += [extra_args for extra_args in os.environ["CMAKE_ARGS"].split(" ") if extra_args]
+
+        # special treatment for Darwin
+        if sys.platform.startswith("darwin"):
+            archs = re.findall(r"-arch (\S+)", os.environ.get("ARCHFLAGS", ""))
+            if archs:
+                cmake_args += ["-DCMAKE_OSX_ARCHITECTURES={}".format(";".join(archs))]
+
+        # setup build arguments
+        build_args = []
+        if sys.platform.startswith("win"):
+            build_args += ["--config", cfg]
+
+        # set number of threads for parallel build
+        if "CMAKE_BUILD_PARALLEL_LEVEL" not in os.environ:
+            if hasattr(self, "parallel") and self.parallel:
+                build_args += [f"-j{self.parallel}"]
+            else:
+                build_args += [f"-j{os.cpu_count()}"]
+
+        # create temporary build directory
+        build_temp = Path(self.build_temp) / ext.name
+        if not build_temp.exists():
+            build_temp.mkdir(parents=True)
+
+        # configure and build
+        subprocess.run(
+            ["cmake", ext.sourcedir, *cmake_args], cwd=build_temp, check=True
+        )
+        subprocess.run(
+            ["cmake", "--build", ".", *build_args], cwd=build_temp, check=True
+        )
+
+
+if __name__ == "__main__":
+    setup(
+        name = "pyopensn",
+        author = "The OpenSn Authors",
+        description = "Python extension for OpenSn",
+        packages = ["pyopensn"],
+        ext_modules = [CMakeExtension("pyopensn.__init__")],
+        cmdclass = {"build_ext": CMakeBuilder},
+        install_requires = ["mpi4py", "numpy"]
+    )
```

Comment:
I will make the comment here, but it is more general that this file only.

If we are building the python binding inside `pyopensn`, I see that `libopensn.0.0.1.dylib`, `libopensn.0.dylib`, and `libopensn.dylib` are also created there.

My preference would be to have the binaries build in the build dir. But then running the python script needs to point there, which may be more difficult to achieve (it depends how the scripts are executed).

Backup plan is to just add:
```
pyopensn/libopensn.0.0.1.dylib
pyopensn/libopensn.0.dylib
pyopensn/libopensn.dylib
```
into `.gitignore` - we don't want people to accidentally stage and commit these files.

Reply:
I see you did actually add `pyopensn/libopensn*.so*` into `.gitignore`, so I guess we just need the `dylib` version of the line